dataflowengineoss/src/main/scala/io/shiftleft/dataflowengineoss/passes/reachingdef/ReachingDefProblem.scala

```diff
@@ -125,29 +120,27 @@ class ReachingDefTransferFunction(method: nodes.Method) extends TransferFunction
   def initKill(method: nodes.Method,
                gen: Map[nodes.StoredNode, Set[Definition]]): Map[nodes.StoredNode, Set[Definition]] = {
 
-    val baseToCalls: Map[TrackedBase, List[nodes.Call]] = method.start.call.l
-      .map { c =>
-        (TrackingPointToTrackedBase(c), c)
+    val baseToCalls: Map[TrackedBase, List[(nodes.Call, AccessPath)]] = method.start.call.l
+      .map { call =>
+        val (base, path) = TrackingPointMethods.toTrackedBaseAndAccessPath(call)
+        (base, (call, path))
       }
       .groupBy(_._1)
       .map { case (k, v) => (k, v.map(_._2)) }
 
     def allOtherInstancesOf(node: nodes.StoredNode): Set[nodes.StoredNode] = {
       node match {
         case call: nodes.Call =>
-          val base = TrackingPointToTrackedBase(call)
-          val accessPath = TrackingPointToAccessPath(call)
-          val otherCallsWithSameBase = baseToCalls
-            .get(base)
-            .toList
-            .flatten[nodes.Call]
-            .filter(_.id != node.id)
-          otherCallsWithSameBase
-            .filter { n =>
-              val (matchResult, _) = TrackingPointToAccessPath(n).matchAndDiff(accessPath.elements)
-              matchResult == MatchResult.EXACT_MATCH
+          val (base, accessPath) = TrackingPointMethods.toTrackedBaseAndAccessPath(call)
```

Comment:
If we only ever want to work with the tracker base and accessPath at once, I'd suggest we wrap them in a class and give that class a descriptive name. I would actually imagine that we can get away with calling the whole thing `AccessPath`.

Reply:
Let's then also introduce a corresponding `apply` method, simply because `TrackingPointMethods.to...` is pretty long.

Reply:
I'm not sure. The object-apply pattern is super ugly for what is effectively a free-standing function.

What I think would be preferred is an implicit, such that we can e.g. `val (base, path) = someNode.trackedBaseAndPath` or `val (base, elems) = someNode.trackedBaseAndPathElems`, where `trackedBaseAndPathElems(node) = {val (base, path) = trackedBaseAndPath(node); (base, path.elements)}`.

The `trackedBaseAndPathElems` would be for the closed tracker -- large parts of it can only deal with `Elements`, not full paths. That's OK in its context, but typing it again and again gets old fast.

The big question about that is whether to expose the implicit to users, i.e. whether to put it into some hidden obscure module (beware of the leopard!), or put it into one of our predef language extension packages.